features/src/main/scala/com/salesforce/op/features/types/Maps.scala

```diff
@@ -291,6 +291,46 @@ object GeolocationMap {
   def empty: GeolocationMap = FeatureTypeDefaults.GeolocationMap
 }
 
+/**
+ * Name representation - map containing information related to a particular name.
+ *
+ * @param value map of keys to values, where keys are one of the following:
+ * - "original"
+ * - "isName"
+ * - "firstName"
+ * - "lastName"
+ * - "gender"
+ */
+class NameStats(value: Map[String, String]) extends TextMap(value) {
+  import NameStats.Keys._
+
+  def isName: Boolean = value.getOrElse(IsNameIndicator, "false") == "true"
+  def isMale: Boolean = value.getOrElse(Gender, "") == "male"
+  def isFemale: Boolean = value.getOrElse(Gender, "") == "female"
+}
+object NameStats {
+  object Keys {
+    val OriginalName = "original"
+    val IsNameIndicator = "isName"
+    val FirstName = "firstName"
+    val LastName = "lastName"
+    val Gender = "gender"
+  }
+  object BooleanStrings {
```

Comment:
what are the boolean strings for?

Reply:
So the `NameStats` type is a `Map[String, String]` (b/c for other values, namely the gender, I need multiple possible values. But then for another key  in `NameStats` the value I want is basically a boolean (see `transformFn` in the companion transformer in `HumanNameDetector`).